cli/packages/gateway/gateway.go

```diff
@@ -282,90 +288,76 @@ func (g *Gateway) registerHeartBeat(ctx context.Context, errCh chan error) {
 	}()
 }
 
-func (g *Gateway) registerRelayIsActive(ctx context.Context, serverAddr string, tlsConf *tls.Config, quicConf *quic.Config, errCh chan error) {
-	ticker := time.NewTicker(5 * time.Second)
-	maxFailures := 3
-	failures := 0
+func (g *Gateway) createPermissionForStaticIps(staticIps string) error {
+	if staticIps == "" {
+		return fmt.Errorf("Missing Infisical static ips for permission")
+	}
 
-	go func() {
-		time.Sleep(2 * time.Second)
-		for {
-			select {
-			case <-ctx.Done():
-				return
-			case <-ticker.C:
-				conn, err := quic.DialAddr(ctx, serverAddr, tlsConf, quicConf)
-				if conn != nil {
-					failures = 0
-					conn.CloseWithError(0, "connection closed")
-				}
+	splittedIps := strings.Split(staticIps, ",")
+	resolvedIps := make([]net.Addr, 0)
+	for _, ip := range splittedIps {
+		ip = strings.TrimSpace(ip)
+		if ip == "" {
+			continue
+		}
 
-				if err != nil && !strings.Contains(err.Error(), "tls: failed to verify certificate") {
-					failures++
-					log.Warn().Err(err).Int("failures", failures).Msg("Relay connection check failed")
+		// if port not specific allow all port
+		if !strings.Contains(ip, ":") {
+			ip = ip + ":0"
+		}
 
-					if failures >= maxFailures {
-						errCh <- fmt.Errorf("relay connection check failed: %w", err)
-					}
-				}
-			}
+		peerAddr, err := net.ResolveUDPAddr("udp", ip)
+		if err != nil {
+			return fmt.Errorf("Failed to resolve static ip for permission: %w", err)
 		}
-	}()
-}
 
-func (g *Gateway) registerPermissionRefresh(ctx context.Context, errCh chan error) {
-	if g.config.InfisicalStaticIp == "" {
-		return
+		resolvedIps = append(resolvedIps, peerAddr)
 	}
 
-	log.Info().Msg("Starting TURN permission refresh routine")
-
-	go func() {
-		ticker := time.NewTicker(30 * time.Second)
-		defer ticker.Stop()
+	if err := g.client.CreatePermission(resolvedIps...); err != nil {
+		return fmt.Errorf("Failed to set ip permission: %w", err)
+	}
+	return nil
+}
 
-		g.refreshPermission(errCh)
+func (g *Gateway) registerRelayIsActive(ctx context.Context, relayAddress string, errCh chan error) error {
+	ticker := time.NewTicker(10 * time.Second)
+	maxFailures := 3
+	failures := 0
 
+	go func() {
+		time.Sleep(2 * time.Second)
 		for {
 			select {
 			case <-ctx.Done():
-				log.Info().Msg("Context cancelled, stopping TURN permission refresh")
 				return
 			case <-ticker.C:
-				g.refreshPermission(errCh)
+				// Configure TLS to skip verification
+				tlsConfig := &tls.Config{
+					InsecureSkipVerify: true,
+					NextProtos:         []string{"infisical-gateway"},
+				}
+				quicConfig := &quic.Config{
+					EnableDatagrams: true,
+				}
+				func() {
+					ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
```

Comment:
use parent ctx instead of background since health check should stop when gateway stops. when there is an error the gateway will shutdown and systemmd should bring it back up. if you wanted it to live forever, then this function should be at a much higher level